proposals/12020-model-registry-integration/README.md

```diff
@@ -0,0 +1,399 @@
+
+### API Design
+
+#### Model Registry Request Structure
+
+```go
+type ModelRegistryRequest struct {
+    ModelName         string            `json:"model_name"`
+    Description       string            `json:"description,omitempty"`
+    ModelFormatName   string            `json:"model_format_name,omitempty"`
+    ModelFormatVersion string           `json:"model_format_version,omitempty"`
+    Owner             string            `json:"owner,omitempty"`
+    Author            string            `json:"author,omitempty"`
+    ContinueOnError   bool              `json:"continue_on_error,omitempty"`
+}
+
+type ModelRegistryConfig struct {
+    URL              string                   `json:"url"`
+    TokenSecretRef   TokenSecretReference     `json:"tokenSecretRef"`
+    CAConfigMapRef   *CAConfigMapReference    `json:"caConfigMapRef,omitempty"`
+    Timeout          string                   `json:"timeout,omitempty"`
+    RetryAttempts    int                      `json:"retryAttempts,omitempty"`
+    TLSVerify        *bool                    `json:"tlsVerify,omitempty"`
+}
+
+type TokenSecretReference struct {
+    SecretName      string `json:"secretName"`
+    SecretNamespace string `json:"secretNamespace,omitempty"`
+    TokenKey        string `json:"tokenKey"`
+}
+```
+
+### Security Considerations
+
+1. **Token Management**: Authentication tokens stored in Kubernetes secrets
+2. **Network Security**: TLS certificate validation for Model Registry connections
+3. **Namespace Isolation**: Configuration scoped to individual namespaces
+4. **Audit Logging**: All registration attempts are logged
+5. **Input Validation**: Limits the fields that can be set by a user
+
+## Test Plan
+
+### Unit Tests
+
+#### Configuration Tests
+
+- Valid configuration parsing
+- Invalid configuration error handling
+- Default value application
+
+#### SDK Tests
+
+- `model.register()` method validation
+- Parameter validation and defaults
+- Metadata serialization
+- Error handling in SDK
+
+### Integration Tests
+
+#### Successful Scenarios
+
+- Model registration with minimal configuration
+- Model registration with full metadata
+
+#### Error Scenarios
+
+- Model Registry API unavailable
+- Invalid authentication token
+- Invalid model metadata
+- Duplicate model version registration
+
+### Graduation Criteria
+
+N/A
+
+## Implementation History
+
+- Initial proposal: 2025-06-27
+
+## Drawbacks
+
+1. **Configuration Overhead**: Requires per-namespace configuration, though this enables proper multi-tenancy
+2. **Dependency on Model Registry**: Creates dependency on external Model Registry service availability
+3. **API Coupling**: Tight coupling to Model Registry API version and structure
+
+## Alternatives
+
+### Alternative 1: Direct SDK Integration
+
+Instead of launcher-based registration, implement direct Model Registry client integration in the SDK. This would expose
+infrastructure details to user code and require authentication handling in components.
```

Comment:
as comment above, this isn't necessarily the case; you can use analogous injection mechanism for the authentication, so that the user can simply focus on using the MR py client.